internal/pkg/runtime/engine/singularity/prepare_linux.go

```diff
@@ -1115,32 +1115,34 @@ func (e *EngineOperations) loadImages(starterConfig *starter.Config) error {
 		// C starter code will position current working directory
 		starterConfig.SetWorkingDirectoryFd(int(img.Fd))
 
-		if err := overlay.CheckLower(img.Path); overlay.IsIncompatible(err) {
-			layer := singularityConfig.UnderlayLayer
-			if !e.EngineConfig.File.EnableUnderlay {
-				sylog.Debugf("Could not fallback to underlay, disabled by configuration ('enable underlay = no')")
-				layer = singularityConfig.DefaultLayer
-			}
-			e.EngineConfig.SetSessionLayer(layer)
-
-			// show a warning message if --writable-tmpfs or overlay images
-			// are requested otherwise make it verbose to not annoy users
-			if e.EngineConfig.GetWritableTmpfs() || len(e.EngineConfig.GetOverlayImage()) > 0 {
-				sylog.Warningf("Fallback to %s layer: %s", layer, err)
-
-				if e.EngineConfig.GetWritableTmpfs() {
-					e.EngineConfig.SetWritableTmpfs(false)
-					sylog.Warningf("--writable-tmpfs disabled due to sandbox filesystem incompatibility with overlay")
+		if e.EngineConfig.GetSessionLayer() == singularityConfig.OverlayLayer {
+			if err := overlay.CheckLower(img.Path); overlay.IsIncompatible(err) {
+				layer := singularityConfig.UnderlayLayer
+				if !e.EngineConfig.File.EnableUnderlay {
+					sylog.Debugf("Could not fallback to underlay, disabled by configuration ('enable underlay = no')")
```

Comment:
While we are looking at this code - I'd generally expect this to be a warning, not a debug message? 

Where we are relying on fallback to underylay for lustre, GPFS etc. then people should probably know clearly that things aren't going to work as they might expect when underlay is disabled?

Reply:
fixed